spec/acceptance/getpwnam_spec.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'function stdlib::getpwnam' do
+  context 'finding the UID of the root user' do
+    let(:pp) do
+      <<-MANIFEST
+      $_password_entry = stdlib::getpwnam('root')
+      file{ '/tmp/roots_uid.txt':
+        ensure  => file,
+        content => "${_password_entry['uid']}\n",
+      }
+
+      # Test the use case for systemd module
+      user{ 'testu':
+        ensure => present,
+        uid    => 250,
+      }
+    MANIFEST
+    end
+
+    it 'works idempotently with no errors' do
+      apply_manifest(pp, catch_failures: true) unless os[:family] == 'windows'
```

Comment:
Instead of repeating the condition (`unless os[:family] == 'windows'`) on each line…